cmd/juju/application/refresher/refresher.go

```diff
@@ -379,16 +378,10 @@ func (r *charmStoreRefresher) Refresh() (*CharmID, error) {
 	newURL, origin, err := r.ResolveCharm()
 	if errors.Is(err, ErrAlreadyUpToDate) {
 		// The charm itself is uptodate but we may need the
-		// URL, origin and macaroon (if there is one)
-		// for updating resources.
-		csMac, csErr := store.AuthorizeCharmStoreEntity(r.authorizer, newURL)
-		if csErr != nil && !strings.Contains(csErr.Error(), "404 NOT FOUND") {
-			return nil, errors.Trace(csErr)
-		}
+		// URL, origin for updating resources.
 		return &CharmID{
-			URL:      newURL,
-			Origin:   origin.CoreCharmOrigin(),
-			Macaroon: csMac,
```

Comment:
CharmID here is defined in refresher/interface.go, Macaroon can be removed from it now too